### PR TITLE
Support the first 16 Ansi256 colors on Windows

### PR DESCRIPTION
**Pull request by kennytm**
Fix the issue described in https://github.com/rust-lang/rust/pull/55788#issuecomment-437632954.

### CODE REVIEW

**Comment by mati865** [comment]
ping @BurntSushi

File: src/lib.rs [hunk 1]
```diff
@@ -1511,17 +1511,10 @@ impl ColorSpec {
         &self,
         console: &mut wincolor::Console,
     ) -> io::Result<()> {
-        use wincolor::Intense;
-
-        let intense = if self.intense { Intense::Yes } else { Intense::No };
-
-        let fg_color = self.fg_color.as_ref().and_then(|c| c.to_windows());
-        if let Some(color) = fg_color {
+        if let Some((intense, color)) = self.fg_color.and_then(|c| c.to_windows(self.intense)) {
```
Comment (BurntSushi) [inline src/lib.rs]: Could you please make sure lines are wrapped to 79 columns inclusive? Similarly for the line below.

(This is why intermediate names were used before. Getting rid of them means you blow the line budget.)

File: src/lib.rs [hunk 2]
```diff
@@ -1569,20 +1564,40 @@ pub enum Color {
...
-            Color::Blue => Some(wincolor::Color::Blue),
-            Color::Green => Some(wincolor::Color::Green),
-            Color::Red => Some(wincolor::Color::Red),
-            Color::Cyan => Some(wincolor::Color::Cyan),
-            Color::Magenta => Some(wincolor::Color::Magenta),
-            Color::Yellow => Some(wincolor::Color::Yellow),
-            Color::White => Some(wincolor::Color::White),
-            Color::Ansi256(_) => None,
-            Color::Rgb(_, _, _) => None,
+    fn to_windows(self, intense: bool) -> Option<(wincolor::Intense, wincolor::Color)> {
```
Comment (BurntSushi) [inline src/lib.rs]: Sorry, but this is a long line too. Try:

```rust
fn to_windows(
    self,
    intense: bool,
) -> Option<(wincolor::Intense, wincolor::Color)> {
```

**Comment by mati865** [comment]
Maybe you should consider adding `rustfmt.toml` with [`max_width` setting](https://rust-lang.github.io/rustfmt/#max_width)?

**Comment by BurntSushi** [comment]
@mati865 I'm aware of rustfmt and its ability to enforce a maximum width setting. I've considered using it, several times over the years, but rustfmt is overall not good enough.

**Comment by BurntSushi** [comment]
This PR is in `termcolor 1.0.5` on crates.io.

**Comment by craigbarnes** [comment]
The 256-color palette was an xterm invention. "ANSI" only ever specified 8 colors. Inventing new terms like `Ansi256` is pushing this long-standing misuse of "ANSI" to insane new levels.

**Pull request merged**